Subject: [PATCH] Disable unsupported MDF native mode without replacing crypto helpers

Keep hostname/certificate checks, AES-GCM helpers and the wildcard table intact.
Return unsupported/failure for hardware MDF APIs; never claim a passed FIPS test.

diff --git a/smali_classes6/com/samsung/android/security/mdf/MdfUtils.smali b/smali_classes6/com/samsung/android/security/mdf/MdfUtils.smali
--- a/smali_classes6/com/samsung/android/security/mdf/MdfUtils.smali
+++ b/smali_classes6/com/samsung/android/security/mdf/MdfUtils.smali
@@ -49,39 +49,6 @@
 .method static constructor blacklist <clinit>()V
     .locals 16
 
-    :try_start_0
-    const-string/jumbo v0, "mdf"
-
-    invoke-static {v0}, Ljava/lang/System;->loadLibrary(Ljava/lang/String;)V
-    :try_end_0
-    .catch Ljava/lang/UnsatisfiedLinkError; {:try_start_0 .. :try_end_0} :catch_0
-
-    goto :goto_0
-
-    :catch_0
-    move-exception v0
-
-    sget-object v1, Ljava/lang/System;->err:Ljava/io/PrintStream;
-
-    new-instance v2, Ljava/lang/StringBuilder;
-
-    const-string v3, "Could not link the library. Error: "
-
-    invoke-direct {v2, v3}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    invoke-virtual {v0}, Ljava/lang/UnsatisfiedLinkError;->getMessage()Ljava/lang/String;
-
-    move-result-object v0
-
-    invoke-virtual {v2, v0}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v2}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object v0
-
-    invoke-virtual {v1, v0}, Ljava/io/PrintStream;->println(Ljava/lang/String;)V
-
-    :goto_0
     const-string/jumbo v14, "or"
 
     const-string/jumbo v15, "org"
@@ -1297,22 +1264,52 @@
     return v1
 .end method
 
-.method public static native blacklist isMdfApplied()Z
+.method public static blacklist isMdfApplied()Z
+    .locals 1
+
+    const/4 v0, 0x0
+
+    return v0
 .end method
 
-.method public static native blacklist isMdfDisabled()Z
+.method public static blacklist isMdfDisabled()Z
+    .locals 1
+
+    const/4 v0, 0x0
+
+    return v0
 .end method
 
-.method public static native blacklist isMdfEnabled()Z
+.method public static blacklist isMdfEnabled()Z
+    .locals 1
+
+    const/4 v0, 0x0
+
+    return v0
 .end method
 
-.method public static native blacklist isMdfEnforced()Z
+.method public static blacklist isMdfEnforced()Z
+    .locals 1
+
+    const/4 v0, 0x0
+
+    return v0
 .end method
 
-.method public static native blacklist isMdfReady()Z
+.method public static blacklist isMdfReady()Z
+    .locals 1
+
+    const/4 v0, 0x0
+
+    return v0
 .end method
 
-.method public static native blacklist isMdfSupported()Z
+.method public static blacklist isMdfSupported()Z
+    .locals 1
+
+    const/4 v0, 0x0
+
+    return v0
 .end method
 
 .method public static blacklist logMdf(Ljava/lang/String;Ljava/lang/String;ZILjava/lang/String;)V
@@ -1505,28 +1502,68 @@
     return-void
 .end method
 
-.method public static native blacklist updateMdfStatus()I
+.method public static blacklist updateMdfStatus()I
+    .locals 1
+
+    const/4 v0, -0x1
+
+    return v0
 .end method
 
-.method public static native blacklist updateMdfVersion()Ljava/lang/String;
+.method public static blacklist updateMdfVersion()Ljava/lang/String;
+    .locals 1
+
+    const/4 v0, 0x0
+
+    return-object v0
 .end method
 
 
 # virtual methods
-.method public native blacklist FIPS_Openssl_SelfTest()I
+.method public blacklist FIPS_Openssl_SelfTest()I
+    .locals 1
+
+    const/4 v0, -0x1
+
+    return v0
 .end method
 
-.method public native blacklist getCCModeFlag()I
+.method public blacklist getCCModeFlag()I
+    .locals 1
+
+    const/4 v0, -0x1
+
+    return v0
 .end method
 
-.method public native blacklist getSBFlag()I
+.method public blacklist getSBFlag()I
+    .locals 1
+
+    const/4 v0, -0x1
+
+    return v0
 .end method
 
-.method public native blacklist setCCModeFlag(I)I
+.method public blacklist setCCModeFlag(I)I
+    .locals 1
+
+    const/4 v0, -0x1
+
+    return v0
 .end method
 
-.method public native blacklist setSBFlagOff()I
+.method public blacklist setSBFlagOff()I
+    .locals 1
+
+    const/4 v0, -0x1
+
+    return v0
 .end method
 
-.method public native blacklist setSBFlagOn()I
+.method public blacklist setSBFlagOn()I
+    .locals 1
+
+    const/4 v0, -0x1
+
+    return v0
 .end method

